code/game/gamemodes/objective.dm

```diff
@@ -290,7 +290,11 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 	return target
 
 /datum/objective/protect/check_completion()
-	return !target || considered_alive(target, enforce_human = human_check)
+	var/obj/item/organ/brain/brain_target
+	if(human_check)
+		brain_target = target.current.getorganslot(ORGAN_SLOT_BRAIN)
+	//Protect will always suceed when someone suicides
```

Comment:
what about nonhuman/noncarbon suicides?

Reply:
what happens if they get, say, soulstoned after they suicide?

Reply:
only humans have this check, and you cannot be soulstoned after suicide iirc